tests/component/task/channels/test_di_channel.py

```diff
@@ -44,3 +44,151 @@ def test___task___add_di_chan_chan_per_line___sets_channel_attributes(
     for chan in chans:
         assert chan.chan_type == ChannelType.DIGITAL_INPUT
         assert chan.di_num_lines == 1
+
+
+@pytest.mark.library_only(
+    reason="Internal method we use for retrieving a channel name isn't supported on gRPC",
+)
```

Comment:
The test cases with `xfail_if_old=False` should pass with gRPC.

Reply:
Good point... what's a good way to detect if the test case is running with grpc interpreter to xfail at runtime?

Reply:
I could just split it into 2 tests, the second batch that fails on grpc is "aggressive abuse" :D

Reply:
grpc_xfail

Reply:
> ...at runtime?

Oh, I see why you did this. 

Delete the `xfail_if_old` param and use `pytest.param` to mark the param data.
https://docs.pytest.org/en/stable/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test

Reply:
https://docs.pytest.org/en/6.2.x/skipping.html#skip-xfail-with-parametrize has another example using skipif to do runtime checks

Reply:
Splitting the test is fine